ocsp/responder.go

```diff
@@ -159,7 +171,20 @@ func (rs Responder) ServeHTTP(response http.ResponseWriter, request *http.Reques
 		return
 	}
 
+	parsedResponse, err := ocsp.ParseResponse(ocspResponse, nil)
+	if err != nil {
+		log.Errorf("Error parsing response: %s", err)
+		response.Write(unauthorizedErrorResponse)
+		return
+	}
+
 	// Write OCSP response to response
+	response.Header().Add("Last-Modified", parsedResponse.ProducedAt.Format(time.RFC1123))
+	response.Header().Add("Expires", parsedResponse.NextUpdate.Format(time.RFC1123))
+	maxAge := int64(parsedResponse.NextUpdate.Sub(rs.clk.Now()) / time.Second)
+	if maxAge > 0 {
+		response.Header().Add("Cache-Control", fmt.Sprintf("max-age=%d", maxAge))
```

Comment:
`Cache-Control` should also include `public,no-transform,must-revalidate`, and the `max-age` should be a bit earlier than `NextUpdate` such that a client observing HTTP cache semantics will refresh before its cached response expires (`NextUpdate`-wise)

See https://tools.ietf.org/html/rfc5019#section-6 for details. Note that it also suggests to include an `ETag` and RECOMMENDS a value for it.